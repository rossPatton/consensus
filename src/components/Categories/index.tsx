import cx from 'classnames';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {categories} from '~app/constants';

const Categories = memo(() => (
  <ul className="d:pl-5 d:pr-5 font-semibold text-left">
    {categories.map((cat, i) => (
      <li
        key={i}
        className="w-full d:pl-5 d:pr-5 mb-3 d:mb-4">
        <div
          className={cx({
            'flex': true,
            'justify-end': cat.align === 'right',
          })}>
          <h2 className="text-2 d:text-1 w-1/2 d:max-w-1/2 mb-2">
            <Link to={`/categories/${cat.slug}`}>
              {cat.display}
            </Link>
          </h2>
        </div>
        <div
          className={cx({
            'flex flex-row': true,
            'flex-row-reverse': cat.align === 'right',
          })}>
          <Link
            to={`/categories/${cat.slug}`}
            className={cx({
              'w-full d:min-w-1/2': true,
              'mr-3': cat.align === 'left',
              'ml-3': cat.align === 'right',
            })}>
            <img
              alt=""
              className="w-full rounded-lg"
              src={`/images/${cat.slug}_sm.svg`}
              width="480"
            />
          </Link>
          <p className="hidden d:block text-2 text-gray-5 leading-loose">
            {cat.description}
          </p>
        </div>
      </li>
    ))}
  </ul>
));

export default Categories;
