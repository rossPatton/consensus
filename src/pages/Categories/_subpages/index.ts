import loadable from '@loadable/component';

export const CategoryList = loadable(() =>
  import(/* webpackChunkName: "CategoryList" */'./CategoryList'),
);

export const Category = loadable(() =>
  import(/* webpackChunkName: "Category" */'./Category'),
);
