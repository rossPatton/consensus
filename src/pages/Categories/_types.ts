import { match } from 'react-router';

export type tProps = {
  match: match & { params: tCategoryParams },
}
