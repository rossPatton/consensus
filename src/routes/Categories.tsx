import loadable from '@loadable/component';

const CategoriesComponent = loadable(() =>
  import(/* webpackChunkName: "Categories" */'~app/pages/Categories'),
);

export const Categories = {
  component: CategoriesComponent,
  exact: true,
  path: '/categories/:category?',
};
