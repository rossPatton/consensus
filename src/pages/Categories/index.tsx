import React from 'react';

import {categoryMap} from '../../constants';
import {ErrorBoundary} from '../../containers';
import {Category, CategoryList} from './_subpages';
import {tProps} from './_types';

const Categories = (props: tProps) => {
  const {match: {params}} = props;
  const renderCategory = !!params.category;

  return (
    <ErrorBoundary>
      <div className="contain mT4">
        <h1 className="fs2 mB3">
          {renderCategory
            ? `Organizations by category: ${categoryMap[params.category]}`
            : 'Categories'}
        </h1>
        {!renderCategory && <CategoryList />}
        {renderCategory && (
          <Category match={props.match} />
        )}
      </div>
    </ErrorBoundary>
  );
};

export default Categories;
