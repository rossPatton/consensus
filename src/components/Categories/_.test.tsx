import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import render from 'react-test-renderer';

import Categories from '.';

describe('components/Categories', () => {
  it('renders as expected', () => {
    const component = render.create(<Router><Categories /></Router>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
