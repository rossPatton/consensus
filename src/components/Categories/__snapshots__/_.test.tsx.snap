// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<ul
  className="d:pl-5 d:pr-5 font-semibold text-left"
>
  <li
    className="w-full d:pl-5 d:pr-5 mb-3 d:mb-4"
  >
    <div
      className="flex"
    >
      <h2
        className="text-2 d:text-1 mb-2 d:max-w-6/12"
      >
        <a
          href="/categories/community"
          onClick={[Function]}
        >
          Community
        </a>
      </h2>
    </div>
    <div
      className="flex flex-row"
    >
      <a
        className="w-full d:min-w-6/12 mr-3"
        href="/categories/community"
        onClick={[Function]}
      >
        <img
          alt="View Community groups"
          className="w-full rounded-lg"
          src="/images/community_sm.svg"
          width="480"
        />
      </a>
      <p
        className="hidden d:block text-2 text-gray-5 leading-loose"
      >
        Your local neighborhood group. From your church, to community centers, any kind of miscellaneous group can be found here.
      </p>
    </div>
  </li>
  <li
    className="w-full d:pl-5 d:pr-5 mb-3 d:mb-4"
  >
    <div
      className="flex justify-end"
    >
      <h2
        className="text-2 d:text-1 mb-2 d:max-w-6/12 w-6/12"
      >
        <a
          href="/categories/cooperative"
          onClick={[Function]}
        >
          Cooperative
        </a>
      </h2>
    </div>
    <div
      className="flex flex-row flex-row-reverse"
    >
      <a
        className="w-full d:min-w-6/12 ml-3"
        href="/categories/cooperative"
        onClick={[Function]}
      >
        <img
          alt="View Cooperative groups"
          className="w-full rounded-lg"
          src="/images/cooperative_sm.svg"
          width="480"
        />
      </a>
      <p
        className="hidden d:block text-2 text-gray-5 leading-loose"
      >
        For your local worker-owned or membership-based cooperative. Come to consensus, on Consensus.
      </p>
    </div>
  </li>
  <li
    className="w-full d:pl-5 d:pr-5 mb-3 d:mb-4"
  >
    <div
      className="flex"
    >
      <h2
        className="text-2 d:text-1 mb-2 d:max-w-6/12"
      >
        <a
          href="/categories/union"
          onClick={[Function]}
        >
          Unions and Labor
        </a>
      </h2>
    </div>
    <div
      className="flex flex-row"
    >
      <a
        className="w-full d:min-w-6/12 mr-3"
        href="/categories/union"
        onClick={[Function]}
      >
        <img
          alt="View Unions and Labor groups"
          className="w-full rounded-lg"
          src="/images/union_sm.svg"
          width="480"
        />
      </a>
      <p
        className="hidden d:block text-2 text-gray-5 leading-loose"
      >
        Find your local union, worker's center or other labor-affilitated group. Never miss another union meeting!
      </p>
    </div>
  </li>
  <li
    className="w-full d:pl-5 d:pr-5 mb-3 d:mb-4"
  >
    <div
      className="flex justify-end"
    >
      <h2
        className="text-2 d:text-1 mb-2 d:max-w-6/12 w-6/12"
      >
        <a
          href="/categories/political"
          onClick={[Function]}
        >
          Political
        </a>
      </h2>
    </div>
    <div
      className="flex flex-row flex-row-reverse"
    >
      <a
        className="w-full d:min-w-6/12 ml-3"
        href="/categories/political"
        onClick={[Function]}
      >
        <img
          alt="View Political groups"
          className="w-full rounded-lg"
          src="/images/political_sm.svg"
          width="480"
        />
      </a>
      <p
        className="hidden d:block text-2 text-gray-5 leading-loose"
      >
        From anarchist collectives to the Democratic Socialists of America or Tech Worker's Coalition, get involved with your local political group today.
      </p>
    </div>
  </li>
</ul>
`;
